src/common/tests/SConscript

```diff
@@ -56,30 +54,32 @@ def scons():
     tenv.Install('$PREFIX/bin/', [common_test])
 
     unit_env = tenv.Clone()
-    # Some syscalls are aliased when fortification is enabled, which makes
-    # mocking in unit tests into a headache.
-    # The compiler decides this on its own, but let's force the issue.
-    compiler_setup.remove_fortify(unit_env)
+    unit_env.AppendUnique(OBJPREFIX='utest_')
+
+    common_mock_ld_script = "%s/common-mock-ld-opts" % Dir('.').srcnode()
+    unit_env.AppendUnique(LINKFLAGS=['-Wl,@%s' % common_mock_ld_script])
+
+    mock_test_utils = unit_env.SharedObject(['test_mocks.c', 'test_utils.c'])
+    drpc_test_utils = unit_env.SharedObject(['../drpc.c']) + mock_test_utils
```

Comment:
You probably need to add something like
Depends(mock_test_utils, common_mock_ld_script)
Depends(drpc_test_utils, common_mock_ld_script)

This will ensure that if you change the script, it should rebuild those programs.  Otherwise, I think it won't happen.  maybe play around with that a bit.

Reply:
Ah, oops!

Reply:
Done. I had to add dependencies for the binaries directly to get the desired behavior.